Completed coffee machine

DVitushkin opened the pull request.

DVitushkin committed:
Add machine exception class

added stub of internal class to work with unexpected behavior

DVitushkin committed:
Add ingredient class

Added ingredient class for manipulaion with drinks and recipe

DVitushkin committed:
[Fix] refactor ingredient class

Simplified Ingredient class and added new inherited class special for coffee machine use

DVitushkin committed:
[Update] custom exception set

Added Ingredient exception class for more detailed report

DVitushkin committed:
Add beverage class

Added beverage class to simlpified work with different recipe of coffee

DVitushkin committed:
Add Machine class with controls and drinks enum

Added class coffee machine with prepared getters and setters
also added entry point for starting coffee machine

DVitushkin committed:
Fix for ingredients class

A little fix because of my carelessness

DVitushkin committed:
Add main menu loader and handler to menu

Added logic for comunicate with user and cooking a  beverage

DVitushkin committed:
Add show recipe method

Added to main menu handler function - action to show recipe
and update action to clean the machine

DVitushkin committed:
Add profile class

Added profile class and profile exception class

DVitushkin committed:
Add logging functional

Added to pom.xml file logging dependency and logging call for positive actions

DVitushkin committed:
Add logging function to negative action

DVitushkin committed:
Add quick fix in imports

MVlasovD commented on src/main/java/org/DVitushkin/App.java:
```diff
@@ -0,0 +1,53 @@
+
+import java.util.ArrayList;
+
+public class App 
+{
```
```suggestion
public class App {
```

MVlasovD commented on src/main/java/org/DVitushkin/machine/Machine.java:
```diff
@@ -0,0 +1,393 @@
+                return ingredient;
+            }
+        }
+        return null;
```
не очень хорошая практика возвращать null, даже если после ифа

MVlasovD commented on src/main/java/org/DVitushkin/machine/Machine.java:
```diff
@@ -0,0 +1,393 @@
+                return beverage;
+            }
+        }
+        return null;
```
!

DVitushkin committed:
Replace return value from null to new object in getByName methods

DVitushkin committed:
Simplified the coffee machine class

Simplified the coffee machine class by adding classes of service to entity like:
- beverage list
- ingredients list
- profile list

DVitushkin committed:
Add Log history service to control log record flow

DVitushkin committed:
Fix logic of communication while create user

DVitushkin committed:
Add log message to check system and turn On/Off methods

MVlasovD commented on src/main/java/org/DVitushkin/App.java:
```diff
@@ -0,0 +1,53 @@
+
+import java.util.ArrayList;
+
+public class App 
+{
```
```suggestion
public class App {
```

DVitushkin committed:
fix code style miss in App file
